### DISCUSSION TITLE
Is conditional column mapping possible?

### DISCUSSION THREAD

**Discussion by maxmac** [linq2db/linq2db#3378] [Q&A] [answered]
I have a scenario in NopCommerce where the base solution needs to know some parts about the plugins installed. The entity model contains columns installed into the database by the plugin, and i want those to be mapped to the database as long as the plugin is installed. 

`[AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = true, Inherited = true)]
    public class PluginSpecificForAttribute : ColumnAttribute
    {
        public PluginSpecificForAttribute(string systemName)
        {
            IsColumn = EngineContext.Current.ResolveAll<IInstalledPluginMarker>().Any(plugin => plugin.SystemName == systemName);
        }
    }`

IsColumn is set to the correct value, but linq2db does not care for me setting the IsColum property in my own attribute. Has anyone else solved this issue with conditional column mapping?

**Comment by MaceWindu** [comment, marked as answer]
We don't have explicit support for mapping attributes inheritance, so I would suspect it doesn't look at your attribute at all.
For cases like this I would use custom metadata provider instead. Check this https://github.com/linq2db/linq2db/issues/2501#issuecomment-696707189

**Comment by maxmac** [comment]
Thank you!
I had to add the following code in NopCommerce in the existing 
`FluentMigratorMetadataReader : IMetadataReader`

`protected virtual T GetAttribute<T>(Type type, MemberInfo memberInfo) where T : Attribute`

```
                if (!memberInfo.GetCustomAttribute<PluginSpecificForAttribute>()?.IsColumn ?? false)
                    return null;
```

And it works!
